common/predictive-text/unit_tests/in_browser/resources/models/simple-dummy.js

```diff
@@ -0,0 +1,115 @@
+/**
+ * While handwritten, this class is designed to mirror the potential results of TypeScript compilation of
+ * model TS source files.
+ */
+(function(){
+  var Model = /** @class */ (function() {
+    function Model() { // implements Model
+    }
+
+    Model.capabilities = {
+      maxLeftContextCodeUnits: 64
+    }
+
+    // A direct import/copy from i_got_distracted_by_hazel.json.
+    Model.futureSuggestions = [
+      [
+        {
+          "transform": {
+            "insert": "I ",
+            "deleteLeft": 0
+          },
+          "displayAs": "I"
+        },
+        {
+          "transform": {
+            "insert": "I'm ",
+            "deleteLeft": 0
+          },
+          "displayAs": "I'm"
+        },
+        {
+          "transform": {
+            "insert": "Oh ",
+            "deleteLeft": 0
+          },
+          "displayAs": "Oh "
+        }
+      ],
+      [
+        {
+          "transform": {
+            "insert": "love ",
+            "deleteLeft": 0
+          },
+          "displayAs": "love"
+        },
+        {
+          "transform": {
+            "insert": "am ",
+            "deleteLeft": 0
+          },
+          "displayAs": "am"
+        },
+        {
+          "transform": {
+            "insert": "got ",
+            "deleteLeft": 0
+          },
+          "displayAs": "got"
+        }
+      ],
+      [
+        {
+          "transform": {
+            "insert": "distracted ",
+            "deleteLeft": 0
+          },
+          "displayAs": "distracted by"
+        },
+        {
+          "transform": {
+            "insert": "distracted ",
+            "deleteLeft": 0
+          },
+          "displayAs": "distracted"
+        },
+        {
+          "transform": {
+            "insert": "a ",
+            "deleteLeft": 0
+          },
+          "displayAs": "a"
+        }
+      ],
+      [
+        {
+          "transform": {
+            "insert": "Hazel ",
+            "deleteLeft": 0
+          },
+          "displayAs": "Hazel"
+        },
+        {
+          "transform": {
+            "insert": "the ",
+            "deleteLeft": 0
+          },
+          "displayAs": "the"
+        },
+        {
+          "transform": {
+            "insert": "a ",
+            "deleteLeft": 0
+          },
+          "displayAs": "a"
+        }
+      ]
+    ];    
+
+    return Model;
+  }());
+
+  // It's a 'dummy' model, so there's no need for extra methods and such within the Model's class definition.
+  LMLayerWorker.loadModel(new models.DummyModel(Model.capabilities, {futureSuggestions: Model.futureSuggestions}));
```

Comment:
Presenting a handwritten prototype of a potential first 'compiled' model file.  Outside of this single line (which may require... 'creative'... insertion via string manipulation, the rest of it corresponds closely to TS compilation results, though with some obvious added comments.

Reply:
I thought about it, but decided against it.  The `LMLayerWorker` has its own scope inside the WebWorker, meaning it has a _lot_ less potential for interference with third-party code, which will have no access to that scope unless our code allows it - and that, by loading it in as a model.